torch/utils/data/dataloader.py

```diff
@@ -72,7 +73,7 @@ def default_collate(batch):
         return torch.LongTensor(batch)
     elif isinstance(batch[0], float):
         return torch.DoubleTensor(batch)
-    elif isinstance(batch[0], str):
```

Comment:
This actually eliminates the need for the first case handled in this function (unicode). Can you please remove it? Also, I think it'd be better to create a `string_classes` tuple that contains `(basestring,)` for Python2 and `(str, bytes)` for Python3. This would add support for `bytes` objects.